unified-codes/apps/data-service/src/app/utils.ts

```diff
@@ -0,0 +1,25 @@
+import { IPaginatedResults, IPaginationParameters } from '@unified-codes/data';
+
+export function getPaginatedResults<T>(params: IPaginationParameters<T>): IPaginatedResults<T> {
+  const { results } = params;
+  const totalResults = results.length;
+  const data = paginateResults(params);
+  const entityCount = data.length;
+
+  return {
+    data,
+    hasMore: totalResults > entityCount,
+    totalResults,
+  };
+}
+
+function paginateResults<T>(params: IPaginationParameters<T>): Array<T> {
+  const { first, offset, results } = params;
+
+  if (first < 1) return [];
+  if (!offset) return results.slice(0, first);
+  // don't let us overflow
+  if (offset === results.length - 1) return [];
+
+  return results.slice(offset, Math.min(results.length, offset + first));
+}
```

Comment:
yes, much nicer. I feel slightly dirty adding in a `utilities` file. still deciding re pulling in all results or handing off to dgraph. I've moved the search to dgraph, and it can handle pagination too. for now though, which way do you think?

Reply:
My thinking on this is to go by [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) and pull everything! I know it's wrong to basically load the front-end up with all the data, so it's a temporary solution, but it's cleaner and simpler, for now. At some point down the line we will probably want to move to pagination at the back-end, but as long as we keep the design OOP it should be easy enough to refactor when the time comes (and I think this approach will end up being the most time effective, as it will be easier to reason about pagination properly when we have more mature functionality). 

What do you think of above?